### PR TITLE
Add a notice about `master` not being production-ready in the README

### PR DESCRIPTION
**Pull request by Calinou**
`master` version of https://github.com/godotengine/godot/pull/52560.

It's not always obvious for new users that they need to switch to `3.x` or `3.3` to compile their own build, if they want something suited for production.

## Preview

https://github.com/Calinou/godot/blob/readme-add-branch-notice/README.md#readme

### CODE REVIEW

**Comment by mhilbrunner** [comment]
Looks good, but IMO a bit wordy for being the very first thing on the frontpage.

Maybe just scratch everything besides *"This is the `master` branch and not production-ready. See [Godot release policy] for more."*
(Then add anything more to that release policy page, if needed.)

**Comment by Riteo** [comment]
> Looks good, but IMO a bit wordy for being the very first thing on the frontpage.

I mean, it's a very important warning. IMO this size is great at being noticed. If it were just a single phrase I think it could get unnoticed by some distracted people in an hurry.

**Comment by vnen** [comment]
The issue I see here is that we have to remember to update this at every release and remove it from the 4.0 branch when it is created.

**Comment by akien-mga** [comment]
Yeah I'm not so fond of having to modify the README at every release for this. There's enough things we have to remember to update when branching off. I think it's part of the normal learning process for new users on GitHub / people new to open source software development that the default branch is not a production branch (which is the most common workflow).

**Comment by Calinou** [comment]
I reworded it to not require changing the notice every time a new Godot version is released.

**Comment by akien-mga** [comment]
> I reworded the notice to not require changing it every time a new Godot version is released.

It still needs to be changed before we tag a stable release in `master` and branch off to `4.0`, `4.1`, etc. So that means removing the notice, committing `4.0-stable` version bump, committing `4.1-dev` version bump, readding the notice.

**Comment by mhilbrunner** [comment]
> 
> 
> > I reworded the notice to not require changing it every time a new Godot version is released.
> 
> It still needs to be changed before we tag a stable release in `master` and branch off to `4.0`, `4.1`, etc. So that means removing the notice, committing `4.0-stable` version bump, committing `4.1-dev` version bump, readding the notice.

Hadn't thought too much of that, yeah, thats indeed not great

**Comment by Calinou** [comment]
Is there a way we could avoid having to change the notice entirely?

**Comment by Riteo** [comment]
> Is there a way we could avoid having to change the notice entirely when a new release is branched off?

Thinking about it, maybe we could describe the stability of each branch, even in a separate file? This way we could just add a warning at the top of README.md talking about this file (eg. BRANCHES.md) and avoid updating the README everytime?
